extensions/artifacts/org.wso2.developerstudio.codenvy.ext.appfactory/src/main/java/org/wso2/developerstudio/codenvy/ext/appfactory/server/client/HTTPAppFactoryClient.java

```diff
@@ -0,0 +1,224 @@
+            }
+
+            try {
+                while ((responseLine = bufferedReader.readLine()) != null) {
+                    responseBuilder.append(responseLine);
+                }
+            } catch (IOException e) {
+                log.error("Error in reading login response", e);
+                throw new AppFactoryServerException("Unable to login user, Details : " + loginRequestParams, e);
+            }
+            successLogin = responseBuilder.toString();
+
+            if ("true".equals(successLogin)) {
+                loginResponse.setLoggedIn(true);
+                loginResponse.setErroneousRequest(false);
+            } else {
+                loginResponse.setLoggedIn(false);
+                loginResponse.setErroneousRequest(true);
+                loginResponse.setErrorType(ErrorType.INVALID_CREDENTIALS);
+            }
+
+            try {
+                EntityUtils.consume(loginResponseEntity);
+            } catch (IOException e) {
+                log.error("Error in closing Entity stream of App Factory login response", e);
+            }
+
+        } else {
+            loginResponse.setLoggedIn(false);
+            loginResponse.setErroneousRequest(true);
+            loginResponse.setErrorType(ErrorType.CONNECTION_FAILURE);
+        }
+
+
+        return loginResponse;
+    }
+
+    /**
+     * Wrap the HTTP client to be used as a secure client
+     *
+     * @param baseClient HttpClient to be wrapped
+     * @param serverURL  Server URL
+     * @return Wrapped HttpClient object
+     */
+    public static HttpClient wrapClient(HttpClient baseClient, String serverURL) throws AppFactoryServerException {
+        SSLContext sslContext;
+        try {
+            sslContext = SSLContext.getInstance("TLS");
+        } catch (NoSuchAlgorithmException e) {
+            log.error("Unable to create an SSL context for TLS provider", e);
+            throw new AppFactoryServerException("Error in wrapping HTTP client", e);
+        }
+        X509TrustManager trustManager = new X509TrustManager() {
+
+            public void checkClientTrusted(X509Certificate[] xcs, String string) throws CertificateException {
+                throw new UnsupportedOperationException("Unsupported trust manager operation invoked");
+            }
+
+            public void checkServerTrusted(X509Certificate[] xcs, String string) throws CertificateException {
+                throw new UnsupportedOperationException("Unsupported trust manager operation invoked");
+            }
+
+            public X509Certificate[] getAcceptedIssuers() {
+                return new X509Certificate[0];
+            }
+        };
+        try {
+            sslContext.init(null, new TrustManager[]{trustManager}, null);
+        } catch (KeyManagementException e) {
+            log.error("Error when initializing context with trust manager", e);
```

Comment:
I was following some conventions in exception handling through each extension module. According to that normally each extension module will have a custom Exception type defined, in this case it’s AppFactoryServerException . If an exception occurred at a certain point a method execution, we would normally log the error and throwing a custom defined exception to upstream in method stack. And in message logging I followed some conventions as below,
MethodX() {
……..
Try {
operationX();
} catch (SpecificExceptionOccurredInOperationX e) {
Log.error(“Specific Error Message Related To OperationX”, e);
Throw new CustomException(“A General Error Message Related to Method X”, e);
}
……….
}
So it could achieve two levels messaging with error stack (one specific to actual failed operation and other one related to the method that executes the operation)
That’s why we are not using same error message in constructor too. What do you think about this approach?